src/darwin/Framework/CHIP/MTRCertificateInfo.h

```diff
@@ -0,0 +1,81 @@
+/**
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import <Matter/MTRCertificates.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class MTRDistinguishedNameInfo;
+
+/**
+ * Exposes Matter-specific information from an operational X.509 certificate.
+ *
+ * Note: This class does not support parsing certifactes related to Device Attestation.
+ */
+@interface MTRCertificateInfo : NSObject <NSCopying>
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ * Initializes the receiver with an operational certificate in Matter TLV format.
+ */
+- (nullable instancetype)initWithTLVBytes:(MTRCertificateTLVBytes)bytes;
```

Comment:
This doesn't need the designated initializer bit in the header?

Reply:
it needs to be in the header if we want to allow sub-classes, but since we don't I prefer to leave it out (as otherwise changing which initialisers are designated becomes an API change)